src/lib.rs

```diff
@@ -4,101 +4,96 @@ pub struct DB<T> {
-///
-/// NB: You will need to be explcit about the liftimes in this struct
 #[derive(Debug, PartialEq, Eq)]
-pub struct DBViewMut<T> {
-    entries: Vec<&mut T>,
+pub struct DBViewMut<'a,T> where T : 'a {
+    entries: Vec<&'a mut T>,
 }
 
 /// Filters a DBView using the the given predicate.
-pub fn filter_one<T, F>(view: &DBView<T>, predicate: F) -> DBView< T>
+pub fn filter_one<'a,T,F>(view: &DBView<'a,T>, predicate: F) -> DBView<'a,T>
     where F: Fn(&T) -> bool
 {
-    unimplemented!()
+    view.select_where(predicate)
 }
 
 /// Filters two DBView structs using the same predicate, producing two separate results. This is
 /// the moral equivalent of doing the two filters separately.
-pub fn filter_two<T, F>(view_a: &DBView<T>,
-                        view_b: &DBView<T>,
-                        predicate: F)
-                        -> (DBView<T>, DBView<T>)
+pub fn filter_two<'a,'b,T,F>(view_a: &DBView<'a,T>,
+                             view_b: &DBView<'b,T>,
+                             predicate: F)
+                             -> (DBView<'a,T>, DBView<'b,T>)
     where F: Fn(&T) -> bool
 {
-    unimplemented!()
+    (view_a.select_where(&predicate), view_b.select_where(&predicate))
 }
 
 impl<T> DB<T> {
     /// Creates a DB from the given list of entries
     pub fn new(data: Vec<T>) -> DB<T> {
-        unimplemented!()
+        DB { data: data }
     }
 
     /// Creates a new DBView containing all entries in `self` which satisfy `predicate`
     pub fn select_where<F>(&self, predicate: F) -> DBView<T>
         where F: Fn(&T) -> bool
     {
-        unimplemented!()
+        DBView { entries: self.data.iter().filter(|t| predicate(t)).collect() }
     }
 
     /// Creates a new DBView containing all entries in `self` which satisfy `predicate`
     pub fn select_where_mut<F>(&mut self, predicate: F) -> DBViewMut<T>
         where F: Fn(&T) -> bool
     {
-        unimplemented!()
+        DBViewMut { entries: self.data.iter_mut().filter(|t| predicate(t)).collect() }
     }
 
     /// Returns a DBView consisting on the entirety of `self`
     pub fn as_view(&self) -> DBView<T> {
-        unimplemented!()
+        self.select_where(|_| true)
     }
 
     /// Returns a DBView consisting on the entirety of `self`
     pub fn as_view_mut(&mut self) -> DBViewMut<T> {
-        unimplemented!()
+        self.select_where_mut(|_| true)
     }
 
     /// Returns the number of entries in the DB
     pub fn len(&self) -> usize {
-        unimplemented!()
+        self.data.len()
     }
 }
 
-impl<T> DBView<T> {
+impl<'a,T> DBView<'a,T> {
     /// Creates a new DBView containing all entries in `self` which satisfy `predicate`
-    pub fn select_where<F>(&self, predicate: F) -> DBView<T>
+    pub fn select_where<F>(&self, predicate: F) -> DBView<'a,T>
         where F: Fn(&T) -> bool
     {
-        unimplemented!()
+        DBView { entries: self.entries.iter().map(|t| *t).filter(|t| predicate(t)).collect() }
```

Comment:
It turns out that this `.map(|t| *t)` is built into the standard library for `Clone` types, as `Iterator::cloned`.